src/tests/class_tests/openms/source/PeakIntegrator_test.cpp

```diff
@@ -848,6 +911,74 @@ START_SECTION(PeakShapeMetrics calculatePeakShapeMetrics(
 }
 END_SECTION
 
+// START_SECTION(gradient_descent)
+// {
+  // std::cout.precision(std::numeric_limits< double >::max_digits10);
+  // double h, mu, sigma, tau;
+  // peakIntegrator.gradient_descent(Sat_RTs, Sat_Ints, h, mu, sigma, tau, true);
+  // std::vector<double> out_xs;
+  // std::vector<double> out_ys;
+  // const bool compute_additional_points { false };
+  // peakIntegrator.emg_vector(Sat_RTs, h, mu, sigma, tau, out_xs, out_ys, compute_additional_points);
+  // std::cout << std::endl;
+  // for (const double x : out_xs)
+  // {
+  //   std::cout << x << ", ";
+  // }
+  // std::cout << std::endl;
+  // for (const double y : out_ys)
+  // {
+  //   std::cout << y << ", ";
+  // }
+// }
+// END_SECTION
+
+START_SECTION(void fitEMGPeakModel(
```

Comment:
The tests look good so far.  As we spoke about during our meeting, we will also want to add in tests for each of the sub functions.  These can be quite simple in nature, but they will be invaluable for debugging and also future implementations.  For example, right now we have all of the parameters for the fitting hard coded.  This is intended.  If in the future we want to expand these functions to its own stand alone class to add additional gradient decent and solver options and additional parameterization, migration will be greatly facilitated by unit tests for each of the sub functions.